bids/layout/layout.py

```diff
@@ -386,23 +401,24 @@ def _init_db(self, database_path=None, reset_database=False):
 
         self._set_session(database_file)
 
-        instance_args = self._sanitize_instance_args()
-
         if not reset_database:
             with open(database_sidecar) as fobj:
                 saved_args = json.load(fobj)
             for k, v in saved_args.items():
-                if instance_args[k] != v:
+                if self.init_args[k] != v:
                     raise ValueError(
-                        "Initialization arguments do not match for database_path:"
-                        " {}".format(database_path)
+                        f"Initialization argument ({k!r}) directories "
+                        f" not match fordatabase_path: {database_path}.\n"
```

Comment:
Typo?

Reply:
@effigies you added this? but we can't do f-strings because we support python 3.5?

Reply:
I added the `!r`. But `{!r}` is still valid in a format string.